app/actors/sufia/interpret_visibility_actor.rb

```diff
@@ -0,0 +1,107 @@
+      attributes = @intention.sanitize_params
+      validate(attributes) && apply_visibility(attributes) && next_actor.create(attributes)
+    end
+
+    # Override CC to ensure our custom 'validate' method is used
+    def update(attributes)
+      # Perform same validation/actions as create()
+      create(attributes)
+    end
+
+    private
+
+      # Validate against selected AdminSet's PermissionTemplate (if any)
+      def validate(attributes)
+        # If AdminSet was selected, look for its PermissionTemplate
+        template = Sufia::PermissionTemplate.find_by(admin_set_id: attributes[:admin_set_id]) if attributes[:admin_set_id].present?
+
+        validate_lease(template) && validate_release_type(template) && validate_visibility(attributes, template) && validate_embargo(attributes, template)
+      end
+
+      # Validate that a lease is allowed by AdminSet's PermissionTemplate
+      def validate_lease(template)
+        # Leases are only allowable if a template doesn't require a release period or have any specific visibility requirement
+        # (Note: permission template release/visibility options do not support leases)
+        return true unless @intention.wants_lease? && template.present? && (template.release_period.present? || template.visibility.present?)
+
+        curation_concern.errors.add(:visibility, 'Lease option is not allowed by permission template for selected AdminSet.')
+        false
+      end
+
+      # Validate the selected release settings against template, checking for when embargoes/leases are not allowed
+      def validate_release_type(template)
+        # It's valid as long as embargo is not specified when a template requires no release delays
+        return true unless @intention.wants_embargo? && template.present? && template.release_no_delay?
+
+        curation_concern.errors.add(:visibility, 'Visibility specified does not match permission template "no release delay" requirement for selected AdminSet.')
+        false
+      end
+
+      # Validate visibility complies with AdminSet template requirements
+      def validate_visibility(attributes, template)
+        # NOTE: For embargo/lease, attributes[:visibility] will be nil (see sanitize_params), so visibility will be validated as part of embargo/lease
+        return true unless template.present? && attributes[:visibility].present?
+
+        # Validate against template's visibility requirements
+        return true if template.valid_visibility?(attributes[:visibility])
+
+        curation_concern.errors.add(:visibility, 'Visibility specified does not match permission template visibility requirement for selected AdminSet.')
+        false
+      end
+
+      # When specified, validate embargo is a future date that complies with AdminSet template requirements (if any)
+      def validate_embargo(attributes, template)
+        return true unless @intention.wants_embargo?
+
+        embargo_date = parse_date(attributes[:embargo_release_date])
```

Comment:
Given the attribute name is `:embargo_release_date`, I would like to see the local variable have the same name as the attribute.